sympy/ntheory/factor_.py

```diff
@@ -518,13 +518,34 @@ def perfect_power(n, candidates=None, big=True, factor=True):
                     pp = n.func(num, den), e
         return pp
 
+    if candidates is not None:
+        candidates = set(candidates)
     n = as_int(n)
     if n < 0:
-        pp = perfect_power(-n)
-        if pp:
+        if candidates is None:
+            pp = perfect_power(-n, big=True, factor=factor)
+            if not pp:
+                return False
+
             b, e = pp
-            if e % 2:
+            e2 = e & (-e)
+            b, e = b ** e2, e // e2
+
+            if e <= 1:
+                return False
+
+            if big or isprime(e):
                 return -b, e
+
+            for p in primerange(3, e + 1):
+                if e % p == 0:
+                    return - b ** (e // p), p
+        odd_candidates = {i for i in candidates if i % 2}
+        if not odd_candidates:
+            return False
+        pp = perfect_power(-n, odd_candidates, big, factor)
+        if pp:
+            return pp
```

Comment:
```
>>> perfect_power((-5)**15)
(-5, 15)
>>> perfect_power((-5)**15, [15])
(5, 15)
```

Reply:
Thank you for pointing this out, this branch should return `(-b, e)` too. 

```python
pp = perfect_power(-n, odd_candidates, big, factor)
if pp:
      b, e = pp
      return -b, e
```